scenes/dialogs/dialogs.gd

```diff
@@ -0,0 +1,19 @@
+extends Control
+
+var DIALOGS = [
+	GLoginDialog,
+	GRegisterDialog
+]
+
+func _ready():
+	pass
+
+func _process(delta):
+	pass
+
+func open_single(dialog) -> void:
+	for other in DIALOGS:
+		if other != dialog:
+			other.close()
+	
+	dialog.open()
```

Comment:
Shouldn't use singletons for this, use a UI stack with instantiated windows instead.

Reply:
It is the same thing if we use `preload()` because they are already the most common scenes and they will be the same.

Reply:
This is about code architecture, not about how preload() works. Singletons make code very hard to read and prone to bugs, so don't use them if you don't have to. A UI stack is a really neat way to keep track of your menus that's easier to read and use than what you're doing here.